legend-sdlc-generation-file/src/test/java/org/finos/legend/sdlc/generation/artifact/TestArtifactGenerationFactory.java

```diff
@@ -0,0 +1,84 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.generation.artifact;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+import org.eclipse.collections.api.map.MutableMap;
+import org.eclipse.collections.impl.utility.LazyIterate;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.ArtifactGenerationExtension;
+import org.finos.legend.engine.protocol.pure.v1.PureProtocolObjectMapperFactory;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.engine.shared.core.deployment.DeploymentMode;
+import org.finos.legend.sdlc.generation.file.GenerationOutput;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestArtifactGenerationFactory
+{
+
+    private PureModelContextData getPureModelContextDataFromPath(String path)
+    {
+        URL url = Objects.requireNonNull(getClass().getClassLoader().getResource(path), "Can't find resource '" + path + "'");
+        try
+        {
+            JsonMapper jsonMapper = PureProtocolObjectMapperFactory.withPureProtocolExtensions(JsonMapper.builder().build());
+            return jsonMapper.readValue(url, PureModelContextData.class);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Unable to build PureModelContextData with path '" + path + "'", e);
+        }
+    }
+
+    @Test
+    public void testArtifactGenerationFactory()
+    {
+        PureModelContextData data = getPureModelContextDataFromPath("ArtifactGenerationFactoryTestData.json");
+        PureModel pureModel = new PureModel(data, null, DeploymentMode.TEST);
+        ArtifactGenerationFactory factory = new ArtifactGenerationFactory(pureModel, data, data.getElements());
+        Assert.assertEquals(2, factory.getExtensions().size());
+        TestSimpleFunctionArtifactGenerationExtension functionArtifactGenerationExtension = LazyIterate.selectInstancesOf(factory.getExtensions(), TestSimpleFunctionArtifactGenerationExtension.class).getFirst();
+        TestSimpleEnumArtifactGenerationExtension enumArtifactGenerationExtension =  LazyIterate.selectInstancesOf(factory.getExtensions(), TestSimpleEnumArtifactGenerationExtension.class).getFirst();
+        MutableMap<ArtifactGenerationExtension, List<ArtifactGenerationResult>> results = factory.generate();
+        Assert.assertEquals(2, results.size());
+        // check enum generation extension
```

Comment:
There's so much in this test that it's hard to read. Let's add some spacing to separate different parts out.
```suggestion
        Assert.assertEquals(2, results.size());
        
        // check enum generation extension
```